### DISCUSSION TITLE
DNS resolution

### DISCUSSION THREAD

**Discussion by ghost** [FreeTubeApp/FreeTube#4381] [Q&A]
For unrelated reasons I had a look at the windows DNS logs but there are no entries for FT.

Why? How is FT doing DNS resolution? How to make FT use the system DNS?

**Comment by absidue** [comment]
When you say no entries for FreeTube do you mean no entries from FreeTube the application or no entries with FreeTube domains? If you are looking for entries with FreeTube domains, you won't find any, because FreeTube runs on your computer, there is no backend server, so it talks directly to the services it needs to talk to.

**Reply by ghost** [reply]
I mean entries from FreeTube the application, i.e. how does it resolve e.g. `youtube.com` without using the system DNS?

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
From https://www.electronjs.org/docs/latest/api/app#appconfigurehostresolveroptions

> Configures host resolution (DNS and DNS-over-HTTPS). By default, the following resolvers will be used, in order:
> 
>     1. DNS-over-HTTPS, if the DNS provider supports it, then
>     2. the built-in resolver (enabled on macOS only by default), then
>     3. the system's resolver (e.g. getaddrinfo).

**Reply by ghost** [reply]
Thanks. 2. doesn't apply here, 3. obviously isn't used, so it has to be 1.
The question now is what DoH provider is used and why it is considered a good privacy measure to allow FT to decide that on its own without user choice, probably using one of the big DoH providers I would never voluntarily use?

**Reply by absidue** [reply]
@c3d1c06c-bf26-477e-b0eb-c50ef4477ba6 The docs seem to indicate that it will check if the provider of your system DNS also supports DNS over HTTPS (the link to the list of known providers is in the Electron docs link that @efb4f5ff-1298-471a-8973-3d47447115dc provided), if yes it will use that, otherwise it will fallback to the system DNS.

**Reply by absidue** [reply]
Also just to make it clear we do not configure anything DNS related, so the Electron defaults apply.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
If u want to see to what domain FT connects u can do the following: Open FT -> Open the dev console -> Go to the network tab -> Hover over `Name` to see what domains FT connects to


https://github.com/FreeTubeApp/FreeTube/assets/73130443/beb1440a-f9f7-4249-8b33-2f00553bc97d

**Reply by ghost** [reply]
This question isn't about what domains FT connects to, but how. Is it using e.g. DoH to circumvent the system DNS or does it manipulate the windows logs or how does it prevent windows from knowing about it's DNS queries?
